hermes/src/main/java/in/raveesh/hermes/receivers/HermesBroadcastReceiver.java

```diff
@@ -1,38 +1,36 @@
 package in.raveesh.hermes.receivers;
 
-import android.content.BroadcastReceiver;
+import android.app.Activity;
+import android.content.ComponentName;
 import android.content.Context;
 import android.content.Intent;
-import android.util.Log;
+import android.support.v4.content.WakefulBroadcastReceiver;
 
-import in.raveesh.hermes.Hermes;
+import in.raveesh.hermes.Util;
 
-public class HermesBroadcastReceiver extends BroadcastReceiver {
+public class HermesBroadcastReceiver extends WakefulBroadcastReceiver {
 
+    private Class intentService;
     public HermesBroadcastReceiver() {
     }
 
-    protected boolean gotRegistrationId = false;
-    protected boolean error = false;
-    protected String registrationID;
+    public void setIntentService(Class intentService) {
+        this.intentService = intentService;
+        Util.log("Setting intentService class to - "+intentService);
+    }
 
     @Override
     public void onReceive(Context context, Intent intent) {
-        registrationID = intent.getStringExtra("registration_id");
-        if (registrationID != null) {
-            gotRegistrationId = true;
-            if (Hermes.CALLBACK != null) {
-                Hermes.CALLBACK.registrationComplete(registrationID);
-            }
-        } else if (intent.getExtras() != null && intent.getExtras().get("error") != null) {
-            error = true;
-            if (Hermes.CALLBACK != null){
-                Hermes.CALLBACK.registrationFailed(intent.getExtras().get("error").toString());
-            }
-            ExponentialBackoffReceiver.attemptRegistration(context, Hermes.getDelay(), Hermes.getSenderId());
-        } else {
-            Log.d(Hermes.TAG, "GCM Message Received");
-
-        }
+
+        if(intentService == null ) throw new UnsupportedOperationException(
+                "Please use setIntentService in the " +
+                        "constructor of your broadcastReceiver class.");
+
+        ComponentName comp = new ComponentName(
```

Comment:
I don't think so there would be a problem with having two intent services, that is why I am using setIntentService, which would be called by the subclass of HermesBroadcastReceiver.

Also, I am doing this because he have to specify the class name which we need to start (and mention it in manifest too.). If I just start HermesIntentService, the user would not receive any update in his subclass intentService (here GcmIntentService), because it wont be registered. So for registering the class properly setIntentService API can be used. This would ensure correct IntentService is called (albeit it has to be registered in manifest too)